docs/getting-started/openshift/index.md

```diff
@@ -0,0 +1,195 @@
+- oc configured to access both hub and spoke clusters.
+- Operator must be installed in [cluster scope](https://argocd-operator.readthedocs.io/en/stable/usage/basics/#cluster-scoped-instance) mode.
+- Apps in any Namespace must be enabled for Hub Cluster.
+- Argocd-agentctl binary (for non-production scenario)
+
+
+## Hub Cluster Setup
+
+Hub cluster will be installed in argocd namespace
+
+Before you install the principal's manifest, you will need to create three
+- Create a TLS secret containing the public CA certificate used by Argo CD-agent components
+- Create a TLS secret containing the certificate and private key used by the principal's gRPC service
+- Create a TLS secret containing the certificate and private key used by the principal's resource proxy
+- Create a secret containing the private RSA key used to sign JWT issued by the principal
+
+For non-production scenarios, you can use the argocd-agentctl CLI's PKI tooling. Please be advised that this is not a production-grade setup, and that important pieces of the PKI, such as the CA's private key, will be stored unprotected on your control plane cluster.
+
+Create the PKI on the principal:
+
+```
+argocd-agentctl --principal-context <control plane context> pki init
+```
+
+Issue the certificate for the principal's gRPC service:
+The gRPC service is the service the agents will connect to. This service usually needs to be exposed to the outside world, or at least from your other clusters where the agents are running on. Typically, this means the service is exposed by a load balancer or a node port.
+
+```
+argocd-agentctl pki issue principal \
+    --principal-context <control plane context> \
+    --ip <ip addresses of principal> \
+    --dns <dns names of principal>
+```
+
+For the --ip and --dns values, you want to specify all addresses and DNS names that the principal will be reachable at for the agents, for example --ip 5.5.5.5 --dns my-principal.example.com. If there is a load balancer in front of the principal, make sure it will pass through traffic to the principal - otherwise, client certificates might not be used for authentication.
+
+Issue the certificate for the principal's resource proxy:
+The principal's resource proxy needs to be reachable by your Argo CD API server (argocd-server), which usually runs in the same cluster as the principal. Typically, this service should not be reachable outside the cluster and is exposed using a Kubernetes Service.
+
+```
+argocd-agentctl pki issue resource-proxy \
+    --principal-context <control plane context> \
+    --ip <ip addresses of resource proxy> \
+    --dns <dns names of principal>
+```
+
+Deploy principal on hub cluster using argocd-operator/gitops-operator using Argo CD CR given below
+
+```
+apiVersion: argoproj.io/v1beta1
+kind: ArgoCD
+metadata:
+  name: argocd
+spec:
+  controller:
+    enabled: false
+  argoCDAgent:
+    principal:
```

Comment:
By default, the principal pod deployed by the operator [uses self-signed certs](https://github.com/argoproj-labs/argocd-operator/blob/ed071b55e5ce2d80ea25d2c9b57aae4ed50c342f/controllers/argocdagent/configmap.go#L309).  I think the TLS certs generated above using the `pki` command will not be used in that case. Should we set `ARGOCD_PRINCIPAL_TLS_SERVER_ALLOW_GENERATE` to false?

Reply:
I think it's okay to have it as it is. we can make an update to this part if we decide user need to update this to disable auto generated tls server.